backend/server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/ServerInfoSynchronizer.java

```diff
@@ -41,6 +44,16 @@ public void synchronize(ServerApi serverApi, SonarLintCancelMonitor cancelMonito
     var serverStatus = serverApi.system().getStatus(cancelMonitor);
     var serverVersionAndStatusChecker = new ServerVersionAndStatusChecker(serverApi);
     serverVersionAndStatusChecker.checkVersionAndStatus(cancelMonitor);
-    storage.serverInfo().store(serverStatus);
+    var isMQRMode = isMQRMode(serverApi.settings(), serverStatus.getVersion(), cancelMonitor);
+    storage.serverInfo().store(serverStatus, isMQRMode);
+  }
+
+  private static boolean isMQRMode(SettingsApi settingsApi, String version, SonarLintCancelMonitor cancelMonitor) {
+    var settingResponse = settingsApi.getGlobalSetting("sonar.multi-quality-mode.enabled", cancelMonitor);
+    if (settingResponse == null) {
+      var serverVersion = Version.create(version);
```

Comment:
We should not forget to take SonarCloud into account. For SC MQR mode is always `true` (for now)

Reply:
Good point!